q2_taxa/plugin_setup.py

```diff
@@ -60,6 +60,116 @@
+                 'taxonomic annotations. Features can be retained in the '
+                 'resulting table by specifying one or more include search '
+                 'terms, and can be filtered out of the resulting table by '
+                 'specifying one or more exclude search terms. If both '
+                 'include and exclude are provided, the inclusion critera '
+                 'will be applied before the exclusion critera. Either '
+                 'include or exclude terms (or both) must be provided.')
+)
+
+plugin.methods.register_function(
+    function=filter_seqs,
+    inputs={
+        'taxonomy': FeatureData[Taxonomy],
+        'sequences': FeatureData[Sequence]
+    },
+    parameters={'include': qiime2.plugin.Str,
+                'exclude': qiime2.plugin.Str,
+                'mode':
+                    qiime2.plugin.Str % qiime2.plugin.Choices(['exact',
+                                                               'contains']),
+                'query_delimiter': qiime2.plugin.Str},
+    outputs=[('filtered_sequences', FeatureData[Sequence])],
+    input_descriptions={
+        'taxonomy': ('Taxonomic annotations for features in the provided '
+                     'feature sequences. All features in the feature '
+                     'sequences must have a corresponding taxonomic '
+                     'annotation. Taxonomic annotations for features that are '
+                     'not present in the feature sequences will be ignored.'),
+        'sequences': 'Feature sequences to be filtered.'},
+    parameter_descriptions={
+        'include': ('One or more search terms that indicate which taxa should '
+                    'be included in the resulting sequences. If providing '
+                    'more than one term, terms should be delimited by the '
+                    'query-delimiter character. By default, all taxa '
+                    'will be included.'),
+        'exclude': ('One or more search terms that indicate which taxa should '
+                    'be excluded from the resulting sequences. If providing '
+                    'more than one term, terms should be delimited by the '
+                    'query-delimiter character. By default, no taxa '
+                    'will be excluded.'),
+        'mode': ('Mode for determining if a search term matches a taxonomic '
+                 'annotation. "contains" requires that the annotation '
+                 'has the term as a substring; "exact" requires that the '
+                 'annotation is a perfect match to a search term.'),
+        'query_delimiter': ('The string used to delimit multiple search terms '
+                            'provided to include or exclude. This parameter '
+                            'should only need to be modified if the default '
+                            'delimiter (a comma) is used in the provided '
+                            'taxonomic annotations.')
```

Comment:
It looks like we really need `Set[Str]` for `include` and `exclude`? Then the interface can work out how to get multiple values and `query_delimiter` doesn't need to exist.

Reply:
Right. This is a hold-over until we have that functionality.